### DISCUSSION TITLE
What makes airflow decides to trigger a "new" task

### DISCUSSION THREAD

**Discussion by punx120** [apache/airflow#33972] [Q&A]
Hi,

I'm struggling to understand what makes airflow decides whether or not to run a new task. For example, let's say I have a dag like this that ran successfully which is set to run daily ('30 4 * * 1-5') with catchup=False:

A -> B

I.e both `A` and `B` are marked successful for today's run. Let's say I add a new task `C` like this:
A -> B -> C

Do you expect `C` to always run right away? or is it more subtle than that? In some dags (which are a bit more complex than that), `C` runs, but in other it doesn't...

Thanks
Sylvain

**Comment by potiuk** [comment]
Not sure what youare asking for. What do you mean by "add task"? Generally speaking if you add a task to existing DAG that already run , the past runs are not re-run automatically. You will need to backfill past runs to do it. But if you have some examples and logs that things are behaving differently, and you can explain some details about those and see differences, then likely opening issue with all details would be a good idea.

**Reply by punx120** [reply]
Past runs are not re-run indeed - i'm talking about the current run.  I'll look at the logs the next time i noticed this behavior.
